src/Microsoft.Extensions.Options/IOptions.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Extensions.Options
     /// Used to retreive configured TOptions instances.
     /// </summary>
     /// <typeparam name="TOptions">The type of options being requested.</typeparam>
-    public interface IOptions<out TOptions> where TOptions : class, new()
+    public interface IOptions<TOptions> where TOptions : class, new()
```

Comment:
Changing IOptions to suddenly start returning different values at different times is definitely unexpected behavior, and honestly this is why I assumed we might not want to change the behavior.  Honestly given this change in behavior, I still think maybe we should consider just adding Get(name) to IOptions, but not break IConfigureOptions (have that derive from IConfigureNamedOptions as even though we are changing the IOptions interface), there's not many implementations of IOptions today, while implementing your own IConfigureOptions was a major extensibility point which we should avoid breaking.